### ISSUE TITLE
consul_keys do not read anything

### ISSUE THREAD

**Issue by evilezh** [hashicorp/terraform#7786]
### Terraform Version

tested on 0.7.0-rc2 and 0.7.0-rc3
### Affected Resource(s)

consul_keys
### Terraform Configuration Files

I run this:

```
variable "tf_path" {
  default = "configuration/terraform/"
}

resource consul_key_prefix tf_variables {
  path_prefix = "${var.tf_path}"
  subkeys {
    region                      = "eu-west-1"
.....
}
```

which places key correctly in place, i can see through UI.
next code would expect to read back: 

```
resource "consul_keys" "tf_variables" {
  key  {
    name = "region"
    path = "configuration/terraform/region"
  }
}

output v1 {
  value = "${consul_keys.tf_variables.var.region}"
}
```

But it gives empty string.

```
tfstate:
  "resources": {
                "consul_keys.tf_variables": {
                    "type": "consul_keys",
                    "primary": {
                        "id": "consul",
                        "attributes": {
                            "datacenter": "eu-west-1",
                            "id": "consul",
                            "key.#": "1",
                            "key.2639784802.default": "",
                            "key.2639784802.delete": "false",
                            "key.2639784802.name": "region",
                            "key.2639784802.path": "configuration/terraform/region",
                            "key.2639784802.value": "",
                            "var.%": "1",
                            "var.region": ""
                        },
                        "meta": {
                            "schema_version": "1"
                        }
                    }
                },
```
### 

Either way ..it should be data source, rather than resource.

**Comment by apparentlymart** [comment]
Hi @evilezh! Sorry for the trouble here.

At least for the last part about it being a data source, that's what #7678 is about, and that'll hopefully be merged before 0.7.0 final.

For the issue of it not reading back: that's a weird issue that I've not seen before, and nothing changed in my patch above that is likely to address whatever issue you had there. Those two configuration snippets are from separate Terrraform configs, right? (as opposed to different parts of the same config that would get applied by the same terraform run)

**Comment by stack72** [comment]
Hi @evilezh / @apparentlymart 

I just merged #7678 FWIW :)

P.

**Comment by stack72** [comment]
I believe this is closed via #7678
